src/lib/int.ts

```diff
@@ -1,15 +1,18 @@
-import { Circuit, Field } from '../snarky';
+import { Circuit, Field, Types } from '../snarky';
 import { CircuitValue, prop } from './circuit_value';
 
 export { UInt32, UInt64, Int64 };
 
-class UInt64 extends CircuitValue {
+class UInt64 extends CircuitValue implements Types.UInt64 {
   @prop value: Field;
   static NUM_BITS = 64;
+  type = 'UInt64' as const;
```

Comment:
I added this field to `UInt64`, to make the type conform to the interface `{ value: Field; type: "UInt64" }`, which is used to signify `UInt64`s in the autogenerated party types.
the intention is to make the `UInt64` interface distinct from the `UInt32` one, to get advanced types in `precondition.ts`to work. this also fixes a couple of bug sources, where `UInt32` could be passed into functions that expected `UIn64` and vice versa

Reply:
Can we use a phantom type here instead of actually polluting the object at runtime? https://dev.to/busypeoples/notes-on-typescript-phantom-types-kg9 You can put the type parameter on the interface to avoid people having to see it when they use Uint64 .

Reply:
See below, I think I solved it nicely